File: test_dir/cpp/src/classes/inheritance.cpp
#include "classes\inheritance.h"
#include "py_str.h"
#include "pypp_util/print.h"

int BaseClass::add(int val) { return a + val; }
int BaseClass::add2(int val) { return a + val; }
int ChildClass::add(int val) { return add2(val); }
int ChildClass::multiply(int val) { return (a * b) * val; }
void class_inheritance_fn() {
    print(PyStr("CLASS INHERITANCE RESULTS:"));
    ChildClass a = ChildClass(2, 2);
    print(a.add(3));
    print(a.multiply(3));
}

File: test_dir/python/src/classes/inheritance.py
class BaseClass:
    def __init__(self, a: int):
        self.a = a

    def add(self, val: int) -> int:
        return self.a + val

    def add2(self, val: int) -> int:
        return self.a + val


class ChildClass(BaseClass):
    def __init__(self, a: int, b: int):
        BaseClass.__init__(self, a)
        self.b = b

    def add(self, val: int) -> int:
        return self.add2(val)

    def multiply(self, val: int) -> int:
        return self.a * self.b * val


def class_inheritance_fn():
    print("CLASS INHERITANCE RESULTS:")
    # basic
    a: ChildClass = ChildClass(2, 2)
    print(a.add(3))
    print(a.multiply(3))
